lend/src/hooks/useAssetPrice.ts

```diff
@@ -0,0 +1,285 @@
+import axios from "axios";
+import { useCallback, useContext, useEffect, useState } from "react";
+import { Moment } from "moment";
+
+import { Assets, AssetsList } from "../store/types";
+import { isProduction } from "../utils/env";
+import {
+  AssetsPriceData,
+  defaultAssetsPriceData,
+  ExternalAPIDataContext,
+} from "./externalAPIDataContext";
+import {
+  COINGECKO_BASE_URL,
+  COINGECKO_CURRENCIES,
+} from "../constants/constants";
+
+const getHistoricalAssetPricesInUSD = async (
+  currencyName: string
+): Promise<{ price: number; timestamp: number }[]> => {
+  const apiURL = `${COINGECKO_BASE_URL}/coins/${currencyName}/market_chart?vs_currency=usd&days=365&interval=daily`;
+
+  const response = await axios.get(apiURL);
+  const { data } = response;
+
+  return data.prices.map(([timestamp, price]: number[]) => ({
+    timestamp,
+    price,
+  }));
+};
```

Comment:
I wonder if this should be exported into the shared dir?

Reply:
its the same as in the shared dir just that the Assets mapping in smaller so was thinking the array queries would be faster